Automattic/wp-calypso
# Domain Management: UI tweaks

This PR introduces some tweaks to help declutter the `Domain Management` screen.

Related to Automattic/nomado-issues#885

## Proposed Changes

* `Manage All Domains` button is removed from the bottom of the page to the top right corner.
* Hide the domains dropdown list when it's disabled. This happens when the site plan does not support setting a custom primary domain or when there are no domains to select from.
* Only show checkboxes in the domains table when Bulk actions make sense (when there are two or more selectable domains).

## Why are these changes being made?

* Checkboxes and bulk actions are irrelevant when there is only domain the user can act on.
* `Manage All Domains` CTA at the bottom of the page was distracting and cluttered when paired with the other CTAs.
* The disabled dropdown and button in the "Primary Site Address" card were adding non actionable elements to the UI.

## Testing Instructions

We need to check the domain management page within a site context ( `/domains/manage/{site-slug}` ) with the following scenarios and make sure the changes mentioned above work as expected:

* A free site with no custom domains.
* A free site with a custom domain that is either registered or mapped.
* A paid site with no custom domains.
* A paid site with at least one custom domain that is either registered or mapped.

## Screenshots

| Before | After |
| --- | --- |
| <img width="1076" alt="image" src="https://github.com/user-attachments/assets/0c50650c-64da-42bf-9182-0da90edfebbf"> | <img width="1062" alt="image" src="https://github.com/user-attachments/assets/853ca03d-e322-47ae-a13b-3fb407155d7b"> |
| <img width="1067" alt="image" src="https://github.com/user-attachments/assets/80a4e8aa-2b58-4587-8b28-08328b66380d"> | <img width="1069" alt="image" src="https://github.com/user-attachments/assets/af87940e-6b84-4d39-b6f7-b22fe17869af"> |
| <img width="1064" alt="image" src="https://github.com/user-attachments/assets/f7381abd-4515-48d8-9c7f-79475771e082"> | <img width="1062" alt="image" src="https://github.com/user-attachments/assets/aba807cf-977f-44c3-b956-27e5082b8a2e"> |
| <img width="1062" alt="image" src="https://github.com/user-attachments/assets/e8a4b7a9-a91a-46b2-a985-b90d1a6a1a3a"> | <img width="1066" alt="image" src="https://github.com/user-attachments/assets/f61266b6-7371-4d90-8465-deefec8e7f6f"> |
